mds_asssignment_1/guestbook/views.py

```diff
@@ -0,0 +1,29 @@
+from django.shortcuts import render
+
+from .forms import EntryForm
+from .models import Entry
+
+
+def home(request):
+    entry_list = Entry.objects.all().filter(visible=True).order_by('date').reverse()
+    if request.method == 'POST':
+        entry_form = EntryForm(request.POST)
+        if entry_form.is_valid():
+            entry = Entry()
+            entry.fname = entry_form.cleaned_data['fname']
+            entry.lname = entry_form.cleaned_data['lname']
+            entry.email = entry_form.cleaned_data['email']
+            entry.comment = entry_form.cleaned_data['comment']
+            entry.save()
+            return render(request, 'entries.html', {
+                'thankyou': entry.fname,
+                'entry_list': entry_list,
```

Comment:
Everything here at the moment is fine and works well.

I just wanted to point out that the way you're doing this could be prone to bugs in the future if you ever introduce code that evaluates the `entry_list` queryset before the new `Entry` is added. Check out the QuerySet documentation for some information about _when_ it will be evaluated:
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#when-querysets-are-evaluated

It's probably best to get your list of Entries after you've added the new one.